cs/src/comm/epoxy-transport/EpoxyTlsConfig.cs

```diff
@@ -0,0 +1,187 @@
+        /// Optional delegate responsible for validating remote certificates.
+        /// May be <c>null</c> to use the default validation.
+        /// </param>
+        /// <param name="encryptionPolicy">
+        /// The encryption policy to use. Defaults to
+        /// <see cref="System.Net.Security.EncryptionPolicy.RequireEncryption">EncryptionPolicy.RequireEncryption</see>
+        /// </param>
+        /// <param name="enabledProtocols">
+        /// The enabled protocols. Defaults to
+        /// <see cref="SslProtocols.Tls12"/>.
+        /// </param>
+        /// <param name="checkCertificateRevocation">
+        /// Whether certificate revocation is checked. Defaults to <c>true</c>.
+        /// </param>
+        protected EpoxyTlsConfig(
+            RemoteCertificateValidationCallback remoteCertificateValidationCallback = null,
+            EncryptionPolicy encryptionPolicy = EncryptionPolicy.RequireEncryption,
+            SslProtocols enabledProtocols = SslProtocols.Tls12,
+            bool checkCertificateRevocation = true)
+        {
+            RemoteCertificateValidationCallback = remoteCertificateValidationCallback;
+            EncryptionPolicy = encryptionPolicy;
+            EnabledProtocols = enabledProtocols;
+            CheckCertificateRevocation = checkCertificateRevocation;
+        }
+
+        /// <summary>
+        /// Gets an optional delegate responsible for validating remote
+        /// certificates.
+        /// </summary>
+        public RemoteCertificateValidationCallback RemoteCertificateValidationCallback { get; }
+
+        /// <summary>
+        /// Gets the encryption policy to use.
+        /// </summary>
+        public EncryptionPolicy EncryptionPolicy { get; }
+
+        /// <summary>
+        /// Gets the enabled protocols.
+        /// </summary>
+        public SslProtocols EnabledProtocols { get; }
+
+        /// <summary>
+        /// Gets a flag indicating whether certificate revocation is checked.
+        /// </summary>
+        public bool CheckCertificateRevocation { get; }
+    }
+
+    /// <summary>
+    /// Represents server-side TLS options.
+    /// </summary>
+    public class EpoxyServerTlsConfig : EpoxyTlsConfig
+    {
+        /// <summary>
+        /// Creates a new instance of the <see cref="EpoxyServerTlsConfig"/>
+        /// class, which is used to configure TLS paramaters for
+        /// <see cref="EpoxyListener"/> instances.
+        /// </summary>
+        /// <param name="certificate">
+        /// The certificate used to identify this server. May not be <c>null</c>.
+        /// </param>
+        /// <param name="clientCertificateRequired">
+        /// Whether clients are required to present a certificate. Defaults to
+        /// <c>false</c>.
+        /// </param>
+        /// <param name="remoteCertificateValidationCallback">
+        /// Optional delegate responsible for validating remote certificates.
```

Comment:
Is "remote certificates" the client certs we're talking about above? Is this how I enforce that the client cert is signed by a particular authority?